dlib/test/smart_pointers.cpp

```diff
@@ -98,37 +98,37 @@ namespace
 
             DLIB_TEST_MSG(counter == 2,counter);
 
-            DLIB_TEST(p1);
```

Comment:
Can you put a cast into the DLIB_TEST macro instead of putting them on each call?